### DISCUSSION TITLE
How can I filter a table once it has loaded?

### DISCUSSION THREAD

**Discussion by apuchitnis** [TanStack/table#3253] [Q&A] [answered]
How can I set a `select` column filter on one of my table's columns once the table has loaded?

I thought this would be straight-forward, but I couldn't work it out myself unfortunately, and I couldn't find any relevant documentation online.

I tried to use `useEffect` as follows, but to no luck:

```
  useEffect(() => {
      var element = document.getElementById('myColumnId');
      if (element != null) {
        if (element.value == '') {
          element.value = 'Yes'
        }
      };
  }, [])
```

**Comment by apuchitnis** [comment, marked as answer]
Figured this out - didn't need to use `useEffect`.

I used [`initialState.filters`](https://react-table.tanstack.com/docs/api/useFilters#table-options) instead.

Very easy 😄

**Comment by maryamWOW** [comment]
Would you please elaborate on your answer? I used intialState.filters as well. but it wasn't applied to my table.

**Reply by apuchitnis** [reply]
See https://github.com/apuchitnis/open-mic-nights/blob/master/src/index.js#L261.
